src/platform/Zephyr/ZephyrConfig.cpp

```diff
@@ -41,11 +41,12 @@ namespace Internal {
     (key);                                                                                                                         \
     static_assert(sizeof(key) <= SETTINGS_MAX_NAME_LEN, "Config key too long: " key)
 
-// Config namespaces
+// Define configuration and counter keys as part of the CHIP_DEVICE_CONFIG_SETTINGS_KEY subtree
+// so that they get erased when KeyValueStoreManagerImpl::DoFactoryReset() is called.
 // clang-format off
-#define NAMESPACE_FACTORY  "chip-fact/"
-#define NAMESPACE_CONFIG   "chip-conf/"
-#define NAMESPACE_COUNTERS "chip-cntr/"
+#define NAMESPACE_FACTORY  CHIP_DEVICE_CONFIG_SETTINGS_KEY "-fct/"
+#define NAMESPACE_CONFIG   CHIP_DEVICE_CONFIG_SETTINGS_KEY "/cfg/"
+#define NAMESPACE_COUNTERS CHIP_DEVICE_CONFIG_SETTINGS_KEY "/ctr/"
```

Comment:
Why the difference between the different subtree prefix patterns mt-fct and mt/cfg, mt/ctr?

```suggestion
#define NAMESPACE_FACTORY  CHIP_DEVICE_CONFIG_SETTINGS_KEY "/fct/"
#define NAMESPACE_CONFIG   CHIP_DEVICE_CONFIG_SETTINGS_KEY "/cfg/"
#define NAMESPACE_COUNTERS CHIP_DEVICE_CONFIG_SETTINGS_KEY "/ctr/"
```

Reply:
Ah, I see that the comment above sort of explains.  But it would be nice to understand why mt-fct _shouldn't_ be erased on factory reset.

Reply:
I thought that the initial idea behind these namespaces was that the factory settings were immutable, but the current code in `CommissionManager.cpp` actually modifies the discriminator to replace it temporarily, so actually erasing all settings might be a safer option. I'll apply the proposed changes.